app/scripts/components/exploration/hooks/use-stac-metadata-datasets.ts

```diff
@@ -43,12 +52,47 @@ function reconcileQueryDataWithDataset(
 
     if (queryData.status === DatasetStatus.SUCCESS) {
       const domain = resolveLayerTemporalExtent(base.data.id, queryData.data);
+
+      // renderParams precedence: Start with sourceParams from the dataset.
+      // If it's not defined, check for the dashboard render configuration in queryData.
+      // If still undefined, check for asset-specific renders using the sourceParams' assets
+      // property.
+      let renderParams = hasValidSourceParams(base.data.sourceParams)
+        ? base.data.sourceParams
+        : undefined;
+
+      if (!renderParams && queryData.data.renders?.dashboard) {
+        renderParams = queryData.data.renders.dashboard;
+      }
+
+      if (
+        !renderParams &&
+        queryData.data.renders?.[base.data.sourceParams?.assets]
+      ) {
+        renderParams = queryData.data.renders[base.data.sourceParams?.assets];
+
+        // The backend sometimes sends `renderParams.rescale` as a nested array, e.g., [[min, max]],
```

Comment:
I realized I may not have fully clarified why flattening is necessary here (please feel free to correct me if I'm off). When Parcel parses the MDX rescale values, they end up in the format [0, 0.1, 0.01, 0.2]. While our UI handles this well (though I haven't encountered multiple rescale values in the GHG Center instance or across our mocked data), the back-end returns rescale values in a [nested format](https://github.com/developmentseed/titiler/blob/dd773230e1d906625f3cdf12933dfe699e952919/src/titiler/core/titiler/core/dependencies.py#L445), like [[0, 0.1], [0.01, 0.2]]. To make sure we safe guard against both formats so that the UI correctly gets only the min,max values it needs, I believe we need to flatten the `rescale` values from the back-end as a precaution.

Reply:
Yeah, it makes sense to follow the existing pattern - But I think our MDX configuration is outdated, and we should change the system to accommodate the multiple bands - let's ticket this separately.